crates/nu-command/src/network/http/get.rs

```diff
@@ -1,204 +1,209 @@
+                "URL",
+                SyntaxShape::String,
+                "the URL to fetch the contents from",
+            )
+            .named(
+                "user",
+                SyntaxShape::Any,
+                "the username when authenticating",
+                Some('u'),
+            )
+            .named(
+                "password",
+                SyntaxShape::Any,
+                "the password when authenticating",
+                Some('p'),
+            )
+            .named(
+                "max-time",
+                SyntaxShape::Int,
+                "timeout period in seconds",
+                Some('m'),
+            )
+            .named(
+                "headers",
+                SyntaxShape::Any,
+                "custom headers you want to add ",
+                Some('H'),
+            )
+            .switch(
+                "raw",
+                "fetch contents as text rather than a table",
+                Some('r'),
+            )
+            .switch(
+                "insecure",
+                "allow insecure server connections when using SSL",
+                Some('k'),
+            )
+            .switch(
+                "full",
+                "returns the full response instead of only the body",
+                Some('f'),
+            )
+            .switch(
+                "allow-errors",
+                "do not fail if the server returns an error code",
+                Some('e'),
+            )
+            .filter()
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Fetch the contents from a URL."
+    }
+
+    fn extra_usage(&self) -> &str {
+        "Performs HTTP GET operation."
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec![
+            "network", "fetch", "pull", "request", "download", "curl", "wget",
+        ]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        run_get(engine_state, stack, call, input)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Get content from example.com",
+                example: "http get https://www.example.com",
+                result: None,
+            },
+            Example {
+                description: "Get content from example.com, with username and password",
+                example: "http get -u myuser -p mypass https://www.example.com",
+                result: None,
+            },
+            Example {
+                description: "Get content from example.com, with custom header",
+                example: "http get -H [my-header-key my-header-value] https://www.example.com",
+                result: None,
+            },
+            Example {
+                description: "Get content from example.com, with custom headers",

```

Comment:
maybe "with multiple custom headers", because the catch here is that a multi-header has syntax
```
-H [key val key val]
```
and not
```
-H [key val] -H [key val]
```
right?

maybe we should even make it clear in the help of `-H` itself? :open_mouth:

Reply:
> maybe we should even make it clear in the help of `-H` itself? 😮

That's a good idea but I wouldn't know how to formulate it promptly this is the current help:
```ruby
 -H, --headers <Any> - custom headers you want to add
 ```

Reply:
yeah the signature is not the best...

maybe this?
```rust
 -H, --headers <Any> - list of custom headers you want to add (`[k v]` or `[k1 v1 k2 v2 ...]`)
```